packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -470,18 +473,48 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		return pendingEntry.handleP.promise;
 	}
 
-	private async uploadBlob(localId: string, blob: ArrayBufferLike): Promise<ICreateBlobResponse> {
+	private async uploadBlob(
+		localId: string,
+		blob: ArrayBufferLike,
+		signal?: AbortSignal,
+	): Promise<ICreateBlobResponse> {
 		return PerformanceEvent.timedExecAsync(
 			this.mc.logger,
 			{ eventName: "createBlob" },
-			async () => this.getStorage().createBlob(blob),
+			async () => this.createBlobInStorage(blob, signal),
 			{ end: true, cancel: this.runtime.connected ? "error" : "generic" },
 		).then(
 			(response) => this.onUploadResolve(localId, response),
-			async (err) => this.onUploadReject(localId, err),
+			async (err) => this.onUploadReject(localId, err, signal),
 		);
 	}
 
+	private async createBlobInStorage(
+		blob: ArrayBufferLike,
+		signal?: AbortSignal,
+	): Promise<ICreateBlobResponse> {
+		return new Promise((resolve, reject) => {
```

Comment:
if you add explicit typing to the Promise here via the generic i think you will find the bug, specifically that you are calling resolve always, and passing it a promise, so i think this promise will return a promise ,or always wait for it.

you will want something more like 
```
Promise.race([
   this.getStorage().createBlob(blob), 
   new Promise<never>((_resolve, reject) =>
					signal.addEventListener("abort", () => reject(new Error(signal.reason))),
				),
]);

```

Reply:
with my above feedback, i don't think we really need anything like this, we just need to stop waiting on the upload promise we already have, there isn't much value in creating a new promise that we also abort at this layer